sdk/cosmosdb/cosmos/src/CosmosClient.ts

```diff
@@ -153,4 +157,23 @@ export class CosmosClient {
   public offer(id: string): Offer {
     return new Offer(this, id, this.clientContext);
   }
+
+
+  /**
+   * Clears background endpoint refresher. Use client.dispose() when destroying the CosmosClient within another process.
+   */
+  public dispose(): void {
+    clearTimeout(this.endpointRefresher)
+  }
+
+  private async backgroundRefreshEndpointList(globalEndpointManager: GlobalEndpointManager, refreshRate: number) {
+    this.endpointRefresher = setInterval(() => {
+      try {
+        globalEndpointManager.refreshEndpointList();
+      } catch (e) {
+        console.warn(`Failed to refresh endpoints:  ${e}`)
```

Comment:
not sure if it makes a difference, but I like to pass the exception as its own arg (you sometimes get better inspection that way in the browser at least):

```suggestion
        console.warn("Failed to refresh endpoints", e);
```